core/include/traccc/utils/particle.hpp

```diff
@@ -0,0 +1,100 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// traccc include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/definitions/qualifiers.hpp"
+#include "traccc/edm/track_parameters.hpp"
+
+// detray include(s).
+#include "detray/definitions/pdg_particle.hpp"
+
+// System include(s).
+#include <stdexcept>
+
+namespace traccc {
+
+namespace detail {
+
+template <typename scalar_t>
+TRACCC_HOST_DEVICE inline detray::pdg_particle<scalar_t>
+particle_from_pdg_number(const std::int32_t pdg_num) {
```

Comment:
That was what with the detray definition. particle type is used in io - see the `io/particle.hpp`
I will change the `int` of particle type to `std::int32_t`

Reply:
This is definitely a subjective comment here. But I/O from/to a text file does not care about the exact bit-width of the type. So `int` is just fine when doing CSV file I/O. The advantage of `std::int32_t` comes if you'd want to write a binary file on Computer A, and then read it on Computer B. In case they define `int` differently (which they would be allowed to do), this could cause an issue. (The same when streaming the variable in a simple binary form over the network.) But text file I/O is basically never a concern with variable types... :thinking:

Reply:
Hmm ok then I will just use int

Reply:
Fixed now